includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2077,6 +1992,9 @@ public function cancel_authorization( $order ) {
 				)
 			);
 		}
+
+		$order->update_meta_data( '_intention_status', $status );
```

Comment:
It looks like we do this in all of the `mark_payment_*` functions (as part of `complete_order_processing`) - is there a case where this might not happen that means we'd need to do it here as well?

Reply:
Yes. The whole method/function this is included in cancels an authorization, and if it is successful, it hands off to the order service to `mark_payment_capture_cancelled`, but if there's some sort of error,[ the next steps are handled there](https://github.com/Automattic/woocommerce-payments/blob/add/order-service/includes/class-wc-payment-gateway-wcpay.php#L1968-L1997). There's an early return after `mark_payment_capture_cancelled` is called so that the meta data isn't updated again if it's called, otherwise the meta data does need to be updated.

Reply:
Thanks for explaining!